event/Event.proto

```diff
@@ -65,6 +65,9 @@ message Event {
   // The end time (in milliseconds) of the event. , measured between the current time and midnight, January 1, 1970 UTC.
   // This field may be empty if the event has not stopped yet, otherwise it should be a valid timestamp after `startTime`.
   int64 endTime = 8;
+  
+  // Name of the layer to which the event belongs. If absent, the OAP would set `layer`=`ID: 0, NAME: undefined`.
```

Comment:
I think undefined is meaningless. Do you expect to ignore no-layer event?

Reply:
> I think undefined is meaningless. Do you expect to ignore no-layer event?

Nope, how about `general`?

Reply:
Do you need that to be general? General service has a clear definition, which is agent installed. Does this fit your case?

In v9 core, the layer is hard required, so `undefined` is just an NPE-protection mechanism, not for a real use case.

Reply:
Currently, there is no layer concept in all events. We can require all new events to have a layer, for old events (reported by old tools / agents), just leave their layers to be empty as before. @wu-sheng correct me if I'm wrong, because the UI can show events of ALL layers that include events without a layer, this is safe, right?

Reply:
For now, event is not integrated into dashboards, but it should be.

I don't have concerns for old events, but when we integrate event, no layer event would be strange and useless.

Reply:
Well, then we can just drop events without layer, and respond an error with message that ask users to upgrade their event report tools, (or disable event reporting) @fgksgf .